### ISSUE TITLE
[master] Bootstrapping build failing in CI

### ISSUE THREAD

**Issue by dseefeld** [dotnet/source-build#1979]
Building the final_sdk in the bootstrap build is failing with the following errors:

```
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Roslyn.Diagnostics.Analyzers/Core/ExportedPartsShouldHaveImportingConstructorCodeFixProvider.cs(19,6): error CS0012: The type 'ExportAttribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Composition.AttributedModel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Roslyn.Diagnostics.Analyzers/Core/TestExportsShouldNotBeDiscoverableCodeFixProvider.cs(18,6): error CS0012: The type 'ExportAttribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Composition.AttributedModel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Roslyn.Diagnostics.Analyzers/Core/ImportingConstructorShouldBeObsoleteCodeFixProvider.cs(19,6): error CS0012: The type 'ExportAttribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Composition.AttributedModel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Utilities/Compiler/HashUtilities.cs(57,85): warning CS8714: The type 'TKey' cannot be used as type parameter 'TKey' in the generic type or method 'ImmutableDictionary<TKey, TValue>'. Nullability of type argument 'TKey' doesn't match 'notnull' constraint. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Utilities/Compiler/HashUtilities.cs(58,85): warning CS8714: The type 'TKey' cannot be used as type parameter 'TKey' in the generic type or method 'ImmutableDictionary<TKey, TValue>'. Nullability of type argument 'TKey' doesn't match 'notnull' constraint. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Utilities/Workspaces/SolutionChangeAction.cs(30,43): warning CS8609: Nullability of reference types in return type doesn't match overridden member. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/FixAnalyzers/FixerWithFixAllAnalyzer.Fixer.cs(19,6): error CS0012: The type 'ExportAttribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Composition.AttributedModel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/AnalyzerReleaseTrackingFix.cs(21,6): error CS0012: The type 'ExportAttribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Composition.AttributedModel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Utilities/Compiler/PooledObjects/PooledDictionary.cs(46,50): warning CS8714: The type 'TKey' cannot be used as type parameter 'TKey' in the generic type or method 'ImmutableDictionary<TKey, TValue>'. Nullability of type argument 'TKey' doesn't match 'notnull' constraint. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/AnalyzerReleaseTrackingFix.FixAllProvider.cs(87,57): warning CS8609: Nullability of reference types in return type doesn't match overridden member. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/AnalyzerReleaseTrackingFix.FixAllProvider.cs(171,57): warning CS8609: Nullability of reference types in return type doesn't match overridden member. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/CompareSymbolsCorrectlyFix.cs(16,6): error CS0012: The type 'ExportAttribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Composition.AttributedModel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/ConfigureGeneratedCodeAnalysisFix.cs(18,6): error CS0012: The type 'ExportAttribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Composition.AttributedModel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/DefineDiagnosticDescriptorArgumentsCorrectlyFix.cs(18,6): error CS0012: The type 'ExportAttribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Composition.AttributedModel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/DefineDiagnosticDescriptorArgumentsCorrectlyFix.CustomFixAllProvider.cs(94,57): warning CS8609: Nullability of reference types in return type doesn't match overridden member. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj]
  /bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/EnableConcurrentExecutionFix.cs(17,6): error CS0012: The type 'ExportAttribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Composition.AttributedModel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/bst/bootstrap_dir/final-sdk/src/roslyn-analyzers.5c7721621bbab897c3ace6bf2c1489dda259eb9e/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj]
```

It seems to have appeared after #1955 was merged bootstrapping.

**Comment by dagood** [comment]
This sounds like what this patch is meant to fix:

0011-Reference-latest-when-running-in-build.patch

```patch
From 20eff0e62e16c49d0b02e861f2cc553d7de66061 Mon Sep 17 00:00:00 2001
From: Davis Goodin <dagood@microsoft.com>
Date: Mon, 28 Dec 2020 11:20:26 -0600
Subject: [PATCH] Reference latest when running in build
...
+  <!--
+    If this project executes during the build (a tool project, rather than a project that compiles
+    against reference assemblies for maximum compatibility), use the latest version of every package
+    rather than whatever the transitive closure happens to include. In source-build, if we use an
+    old reference-only package, it will fail to execute becuase methods are missing implementations.
+  -->
+  <ItemGroup Condition="'$(ExecutesDuringBuild)' == 'true'">
+    <ExecuteDuringBuildLatestPackageId Include="Microsoft.Bcl.AsyncInterfaces" />
+    <ExecuteDuringBuildLatestPackageId Include="System.Collections.Immutable" />
+    <ExecuteDuringBuildLatestPackageId Include="System.Composition" />
+    <ExecuteDuringBuildLatestPackageId Include="System.Composition.AttributedModel" />
+    <ExecuteDuringBuildLatestPackageId Include="System.Composition.Convention" />
+    <ExecuteDuringBuildLatestPackageId Include="System.Composition.Hosting" />
+    <ExecuteDuringBuildLatestPackageId Include="System.Composition.Runtime" />
+    <ExecuteDuringBuildLatestPackageId Include="System.Composition.TypedParts" />
+    <ExecuteDuringBuildLatestPackageId Include="System.Reflection.Metadata" />
+    <ExecuteDuringBuildLatestPackageId Include="System.Runtime.CompilerServices.Unsafe" />
+    <ExecuteDuringBuildLatestPackageId Include="System.Text.Encoding.CodePages" />
```

From the 5.0.102 PR: https://github.com/dotnet/source-build/pull/1975

**Comment by dagood** [comment]
Will be fixed by merging https://github.com/dotnet/source-build/pull/1975.

**Comment by dseefeld** [comment]
As mentioned above, fixed by #1975
